src/Products/Products.php

```diff
@@ -39,28 +39,106 @@ public function delete(string $id) {
+           require_once('../src/views/index.html');
+          }
+      } elseif ($this->checkAlertIsAccepted() == false) {
+        $this->displayAlert();
+        require_once('../src/views/index.html');
+      }
+  }
+    public function beers(){
+      if ($this->checkAlertIsAccepted() == true) {
+        if ($this->checkUserIsLogged() == true) {
+          if ($this->checkThemeIsDark() == true) {
+              require_once('../src/views/beers_logged_dark.php');
+          } elseif ($this->checkThemeIsDark() == false) {
+              require_once('../src/views/beers_logged.php');
+          }
+        } elseif ($this->checkUserIsLogged() == false) {
+           require_once('../src/views/beers.php');
+          }
+      } elseif ($this->checkAlertIsAccepted() == false) {
+        $this->displayAlert();
+        require_once('../src/views/index.html');
+      }
+  }
+    public function view(){
+      if ($this->checkAlertIsAccepted() == true) {
+        if ($this->checkUserIsLogged() == true) {
+          if ($this->checkThemeIsDark() == true) {
+              require_once('../src/views/view_logged_dark.php');
+            } elseif ($this->checkThemeIsDark() == false) {
+              require_once('../src/views/view_logged.php');
+          }
+        } elseif ($this->checkUserIsLogged() == false) {
+           require_once('../src/views/view.php');
+          }
+        } elseif ($this->checkAlertIsAccepted() == false) {
+          $this->displayAlert();
+          require_once('../src/views/index.html');
+      }
+  }
+    public function about(){
+      if ($this->checkAlertIsAccepted() == true) {
+        if ($this->checkUserIsLogged() == true) {
+          if ($this->checkThemeIsDark() == true) {
+              require_once('../src/views/about_logged_dark.php');
+            } elseif ($this->checkThemeIsDark() == false) {
+              require_once('../src/views/about_logged.php');
+          }
+        } elseif ($this->checkUserIsLogged() == false) {
+           require_once('../src/views/about.html');
+          }
+        } elseif ($this->checkAlertIsAccepted() == false) {
+          $this->displayAlert();
+          require_once('../src/views/index.html');
+      }
+  }
+    public function basket(){
+      if ($this->checkAlertIsAccepted() == true) {
+        if ($this->checkUserIsLogged() == true) {
+          if ($this->checkThemeIsDark() == true) {
+              require_once('../src/views/basket_logged_dark.php');
+            } elseif ($this->checkThemeIsDark() == false) {
+              require_once('../src/views/basket_logged.php');
+          }
+        } elseif ($this->checkUserIsLogged() == false) {
+           require_once('../src/views/basket.html');
+          }
+        } elseif ($this->checkAlertIsAccepted() == false) {
+          $this->displayAlert();
+          require_once('../src/views/index.html');
+      }
     }
 
+    public function sum(){
+      if ($this->checkAlertIsAccepted() == true) {

```

Comment:
```php
if ($this->checkAlertIsAccepted() == true) {
 +        if ($this->checkUserIsLogged() == true) {
 +          if ($this->checkThemeIsDark() == true) {
```

to sie powtarza ciągle - wywalić do jednej metody i nie powtarzać się ( DRY - dont repeat youself ).